### PR TITLE
Treeview component

### PR DESCRIPTION
**Pull request by dheeraja00**

### CODE REVIEW

File: src/components/TreeView/TreeNode.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  renderNode = (): any => {
+    const { label = '' } = this.props;
+
+    return label;
+  }
+
+  render() {
+    // If component prop is not present then check for label & render that
+    const { component = this.renderNode } = this.props;
```
Comment (harsimranb) [inline src/components/TreeView/TreeNode.tsx]: @dheeraja00 Can you help me understand the syntax here? `{ component = this.renderNode }` if component is null, then it sets it to renderNode, else it doesn't.  Is this correct?

File: src/components/TreeView/TreeView.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+          this.setState({ source });
+          return true;
+        }
+      }
+    });
+  }
+
+  // Set child nodes active or inactive status
+  toggleChildNode = (source: SourceData[], id: number) => {
+    return source.some((item: SourceData): any => {
```
Comment (harsimranb) [inline src/components/TreeView/TreeView.tsx]: I believe you don't have to specify `: any` here.  If you do, i think it should be a boolean.  Let's not use `any` unless we absolutely have to.

File: src/components/TreeView/TreeView.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+      // Recursive if child node contains another child node
+      if (item.children) {
+        this.toggleChildNode(item.children, id);
+      }
+    });
+  }
+
+  // Render single node, also iterate through its children & render those as well if its parent active status is true
+  renderNode = (item: SourceData): any => {
```
Comment (harsimranb) [inline src/components/TreeView/TreeView.tsx]: Same with `:any` here as before.

File: src/components/TreeView/interface.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { ReactElement } from 'react';
+
+export interface SourceData {
```
Comment (harsimranb) [inline src/components/TreeView/interface.ts]: Good work!

File: src/components/TreeView/interface.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  id: number;
+  label?: string;
+  component?(): ReactElement<any>;
```
Comment (harsimranb) [inline src/components/TreeView/interface.ts]: @dheeraja00 Why is it better to make `component` a function?

**Comment by dheeraja00** [comment]
@harsimranb I have made the required changes please check

**Pull request merged**